src/include/platform/internal/GenericPlatformManagerImpl_POSIX.ipp

```diff
@@ -34,29 +34,30 @@
 
 #include <system/SystemLayer.h>
 
-#include <poll.h>
 #include <assert.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <poll.h>
 #include <sched.h>
-#include <unistd.h>
 #include <sys/select.h>
+#include <unistd.h>
 
 #define DEFAULT_MIN_SLEEP_PERIOD (60 * 60 * 24 * 30) // Month [sec]
 
 namespace chip {
 namespace DeviceLayer {
 namespace Internal {
 
-// Fully instantiate the generic implementation class in whatever compilation unit includes this file.
-template class GenericPlatformManagerImpl_POSIX<PlatformManagerImpl>;
-
 template <class ImplClass>
 CHIP_ERROR GenericPlatformManagerImpl_POSIX<ImplClass>::_InitChipStack(void)
 {
     CHIP_ERROR err = CHIP_NO_ERROR;
 
+#if defined(__APPLE__)
```

Comment:
Is this the right portability check? Is `PTHREAD_RECURSIVE_MUTEX_INITIALIZER` a preprocessor definition? If so, maybe use:

```
#if defined(PTHREAD_RECURSIVE_MUTEX_INITIALIZER)
    mChipStackLock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER;
#elif defined(PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP)
    mChipStackLock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
#else
#error "No defined static initializer for POSIX Threads recursive mutex!"
#endif // defined(PTHREAD_RECURSIVE_MUTEX_INITIALIZER)
```

Reply:
@vivien-apple ?

Reply:
Done.